src/esp/io/URDFParser.h

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+// Code adapted from Bullet3/examples/Importers/ImportURDFDemo ...
+
+#pragma once
+#include <Magnum/Magnum.h>
+#include <Magnum/Math/Color.h>
+#include <Magnum/Math/Matrix4.h>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace tinyxml2 {
+class XMLElement;
+};
+
+////////////////////////////////////
+// Utility/storage structs
+////////////////////////////////////
+
+namespace esp {
+namespace io {
+
+struct UrdfMaterialColor {
+  Magnum::Color4 m_rgbaColor;
+  Magnum::Color3 m_specularColor;
+  UrdfMaterialColor()
+      : m_rgbaColor(0.8, 0.8, 0.8, 1), m_specularColor(0.4, 0.4, 0.4) {}
+};
+
+struct UrdfMaterial {
+  std::string m_name;
+  std::string m_textureFilename;
+  UrdfMaterialColor m_matColor;
+
+  UrdfMaterial() {}
+};
+
+enum UrdfJointTypes {
+  URDFRevoluteJoint = 1,
+  URDFPrismaticJoint,
+  URDFContinuousJoint,
+  URDFFloatingJoint,
+  URDFPlanarJoint,
+  URDFFixedJoint,
+  URDFSphericalJoint,
+
+};
+
+enum UrdfGeomTypes {
+  URDF_GEOM_SPHERE = 2,
+  URDF_GEOM_BOX,
+  URDF_GEOM_CYLINDER,
+  URDF_GEOM_MESH,
+  URDF_GEOM_PLANE,
+  URDF_GEOM_CAPSULE,  // non-standard URDF
+  URDF_GEOM_UNKNOWN,
+};
+
+struct UrdfGeometry {
+  UrdfGeomTypes m_type;
+
+  double m_sphereRadius;
```

Comment:
I'm too lazy to scan this PR to check that this gets properly initialized everywhere :), but in general, PODs like doubles that don't get default initialized make me nervous.